domain/secret/service/consume.go

```diff
@@ -12,37 +12,57 @@ import (
 	secreterrors "github.com/juju/juju/domain/secret/errors"
 )
 
-func (s *SecretService) GetSecretConsumer(ctx context.Context, uri *secrets.URI, unitName string) (*secrets.SecretConsumerMetadata, error) {
-	return nil, secreterrors.SecretConsumerNotFound
-	/*
-		consumerMetadata, err := getConsumerMetadata(...)
-		//if consumerMetadata.Label != "" {
-		//	return consumerMetadata, nil
-		//}
-		// We allow units to access the application owned secrets using the application owner label,
-		// so we copy the owner label to consumer metadata.
-		//md, err := s.getAppOwnedOrUnitOwnedSecretMetadata(uri, "")
-		//if errors.Is(err, errors.NotFound) {
-		//	// The secret is owned by a different application.
-		//	return consumerMetadata, nil
-		//}
-		//if err != nil {
-		//	return nil, errors.Annotatef(err, "cannot get secret metadata for %q", uri)
-		//}
-		//consumerMetadata.Label = md.Label
+// GetSecretConsumerAndLatest returns the secret consumer info for the specified unit and secret, along with
+// the latest revision for the secret.
+// If the unit does not exist, an error satisfying [uniterrors.NotFound] is returned.
+// If the secret does not exist, an error satisfying [secreterrors.SecretNotFound] is returned.
+// If there's not currently a consumer record for the secret, the latest revision is still returned,
+// along with an error satisfying [secreterrors.SecretConsumerNotFound].
+func (s *SecretService) GetSecretConsumerAndLatest(ctx context.Context, uri *secrets.URI, unitName string) (*secrets.SecretConsumerMetadata, int, error) {
+	consumerMetadata, latestRevision, err := s.st.GetSecretConsumer(ctx, uri, unitName)
+
+	if err != nil {
+		return nil, latestRevision, errors.Trace(err)
+	}
+	if consumerMetadata.Label != "" {
+		return consumerMetadata, latestRevision, nil
+	}
+	// We allow units to access the application owned secrets using the application owner label,
+	// so we copy the owner label to consumer metadata.
+	md, err := s.getAppOwnedOrUnitOwnedSecretMetadata(ctx, uri, unitName, "")
+	if errors.Is(err, secreterrors.SecretNotFound) {
+		// The secret is owned by a different application.
```

Comment:
More detailed: 		// The secret is owned by a different application, the unit is consumer.